### PR TITLE
Yanxi/build route for a team

### PR DESCRIPTION
**Pull request by cici2323**

### CODE REVIEW

File: app/basketball/[competition]/teams/[teamid]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,701 @@
+// // import Custom404 from "@/components/404";
```
Comment (umliuxin) [inline app/basketball/[competition]/teams/[teamid]/page.tsx]: on purpose?

File: package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "@types/react-dom": "18.2.4",
     "autoprefixer": "10.4.14",
     "framer-motion": "^10.16.1",
-    "next": "^14.0.4",
+    "next": "^14.2.5",
```
Comment (umliuxin) [inline package.json]: maybe revert this one. 
we don't want the dependency bump along with the feature dev, unless necessary

**Comment by cici2323** [comment]
Solved.

File: app/basketball/[competition]/teams/[teamid]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+  }
+
+  console.log("获取的赛季信息: ", season);
+  console.log("Team ID: ", teamid);
+
+  // 2. 获取球队详细信息
+  let teamResponse;
+  try {
+    const teamUrl = `/basketball/seasonteam?seasonid=${season.id}&teamid=${teamid}`;
+    console.log("请求球队信息的 URL: ", teamUrl);
```
Comment (umliuxin) [inline app/basketball/[competition]/teams/[teamid]/page.tsx]: remove all the console.log?

File: components/basketball/teams/TeamInformation.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
+"use client"; // 指定这是一个客户端组件
+
+import React, { useState } from "react";
+import { Card, CardBody, CardHeader, Image } from "@nextui-org/react";
+import Custom404 from "@/components/404";
+
+interface Team {
```
Comment (umliuxin) [inline components/basketball/teams/TeamInformation.tsx]: we should use `BbTeam` as type

File: app/basketball/[competition]/teams/[teamid]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+                </div>
+
+                {/* 显示球队的赛季表现 (Team Rank Table) */}
+                <div className="bg-gray-100 p-4 rounded shadow mt-4">
+                  <h2 className="text-lg font-bold mb-2 border-l-4 border-lime-600 ps-2">
+                    球队赛季表现
+                  </h2>
+
+                  {/* 判断如果没有数据，则显示提示信息 */}
+                  {teamRank.length > 0 ? (
```
Comment (umliuxin) [inline app/basketball/[competition]/teams/[teamid]/page.tsx]: this should be moved to a component

File: app/basketball/[competition]/teams/[teamid]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+
+                {/* 显示球队队员赛季数据 */}
+                <div className="bg-gray-100 p-4 rounded shadow mt-4">
+                  <h2 className="text-lg font-bold mb-2 border-l-4 border-lime-600 ps-2">
+                    球队队员赛季数据
+                  </h2>
+
+                  {/* 表格外部增加滚动容器 */}
+                  <div style={{ maxHeight: "150px", overflowY: "auto" }}>
+                    <table className="min-w-full bg-white border-collapse">
```
Comment (umliuxin) [inline app/basketball/[competition]/teams/[teamid]/page.tsx]: prefer to use nextui/table

**Review by umliuxin — COMMENTED** [review]
1. could you add some more info and screenshot in the PR
2. for the page.tsx, let's split the large file to several reusable components

**Comment by cici2323** [comment]
Screenshot 2024-10-03 at 5 30 50 PM

**Comment by cici2323** [comment]
This is what the team page looks like：
Screenshot 2024-10-03 at 5 33 00 PM
The features include:
(1) The page currently shows a list of players from the team on the left side. Each player’s photo and name are linked to their individual profile page, and the player list is scrollable.
Screenshot 2024-10-03 at 5 39 59 PM
(2) The top right side of the page shows the team's information, and the email is now linked.
Screenshot 2024-10-03 at 5 44 52 PM
(3) The bottom right side of the page shows the team's season performance and player stats for the season.
Screenshot 2024-10-03 at 5 47 54 PM

**Review by umliuxin — DISMISSED** [review]

**Pull request merged**